src/routing_table/authority.rs

```diff
@@ -96,6 +97,21 @@ impl<N: Xorable + Clone + Copy + Binary + Default> Authority<N> {
     }
 }
 
+impl Authority<XorName> {
+    /// provide the name mathching a single node's public key
+    pub fn single_signing_name(&self) -> Option<&XorName> {
+        match *self {
+            Authority::ClientManager(_)
+            | Authority::NaeManager(_)
+            | Authority::Section(_)
+            | Authority::PrefixSection(_)
+            | Authority::NodeManager(_) => None,
```

Comment:
wondering that in case of other authority, shall we log_or_panic directly?

Reply:
That does not seem right, I actually added a test task to cover sending messages that are not correctly signed. This seems more about sanitizing user input, and so should not panic (since we want to test it).